envbuilder.go

```diff
@@ -108,6 +159,32 @@ func Run(ctx context.Context, opts options.Options) error {
 		}
 	}() // best effort
 
+	runtimeData := runtimeDataStore{
+		ContainerEnv: make(map[string]string),
+		RemoteEnv:    make(map[string]string),
+	}
```

Comment:
Review: Decided to move all variables with larger scope to runtimeData, even though we only serialize container user into e.g. magic image. This is to make it more clear that some variables affect a greater scope, and allows us to share logic between both build and start from cached image.